app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,72 @@
+class AddressesController < ApplicationController
+  before_action :auth_check
+  before_action :load_user, only: %i( edit destroy update)
+
+  def new
+    @user = Current.user 
+    @address = @user.addresses.new
+  end
+
+  def create
+    @address = Current.user.addresses.new(address_params)
+    if @address.save
+      redirect_to user_path(Current.user)
+      flash[:notice] = "Address added Succesfully."
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+    begin
+    rescue StandardError => e
+      redirect_to root_path, status: 303
+      flash[:notice] = "Something went Wrong => .#{e}"
+    end
+  end
+
+   def update
+    begin
+      if @address.present?
+        if @address.update(address_params)
+          redirect_to user_path(@user)
+           flash[:notice] = "Address updated succesfully."
+        else
+          render :edit, status: :unprocessable_entity
+        end
+      end
+    rescue StandardError => e
+      redirect_to root_path, status: 303
+      flash[:notice] = "Something went Wrong.#{e}"
+    end
```

Comment:
If you are going to use this logic in each action then you should find a better and optimized way to do this.